.baseline/checkstyle/checkstyle.xml

```diff
@@ -281,16 +280,13 @@
         </module>
         <module name="InnerAssignment"/> <!-- Java Coding Guidelines: Inner assignments: Not used -->
         <module name="MemberName"> <!-- Java Style Guide: Non-constant field names -->
-            <property name="format" value="^[a-z][a-zA-Z0-9]+$"/>
-            <message key="name.invalidPattern" value="Member name ''{0}'' must match pattern ''{1}''."/>
+            <property name="format" value="^[a-z][a-zA-Z0-9]++$"/>
```

Comment:
Could you explain what the ++ syntax means?

Reply:
The `*+` and `++` are _possessive_ quantifiers (as opposed to `*` and `+` which are _greedy_; and `*?` and `+?` which are `lazy`). 

I found the explanation on this page very useful: https://www.rexegg.com/regex-quantifiers.php#possessive

My understanding is that in cases when we can use both _greedy_ and _possessive_ quantifiers, _possessive_ are faster. Quote form the link above:

> This behavior is particularly useful when you know there is no valid reason why the engine should ever backtrack into a section of matched text, as you can save the engine a lot of needless work.

This seems a common regex feature, but honestly speaking I wasn't familiar with in until @findepi pointed it out (thx for that) in the review of #10673.

I understand using _possessive_ quantifiers is good for performance, OTOH it's possible that many people are unfamiliar with this syntax, so it may be a tradeoff in terms of readability (and also a deviation from checkstyle standards). WDYT?

Reply:
Well checkstyle currently takes less than 1 second on my machine :) so I'm not super worried about the perf. In reading your article it does seem like this is a situation where a good regex engine wouldn't backtrack anyway because $ is incompatible with the previous matching class. Is Java a good regex engine? I don't know.

Anyway, I have no problem with merging this since it is semantically the same for our purposes.